babel.config.js

```diff
@@ -129,7 +125,9 @@ module.exports = function getBabelConfig(api) {
   }
 
   if (process.env.NODE_ENV === 'production') {
-    plugins.push(...productionPlugins);
+    if (!process.env.E2E_BUILD) {
+      plugins.push(['babel-plugin-react-remove-properties', { properties: ['data-testid'] }]);
```

Comment:
What causes this to suddenly be necessary? If possible, would be nice to keep parity between babel configs across repos. With the hope of some day just embedding it in our build tooling.

Reply:
e2e tests build the source-code using this configuration and they rely on the `data-testid`

If this plugin is enabled they simply don't work.

Reply:
For more context, previously it wasn't an issue because in unit tests we were using `data-mui-test` but e2e were using `data-testid`, so the `data-mui-test` could be removed regardless of build. 

But since now `both` use `data-testid`, the issue appears

Reply:
interestingly, this rename in core was done without altering the babel config https://github.com/mui/material-ui/pull/23498

Reply:
I noticed `material` doesn't remove the `data-testid` from the build, that is why they don't have this problem. 

But most of the times they add `data-testid` in the test themselves, not the source code.